### PR TITLE
Init file

### PR DESCRIPTION
**Pull request by Vlaszaty**
Added init script for upstart initstyle

### CODE REVIEW

File: manifests/service.pp [hunk 1]
```diff
@@ -31,6 +36,13 @@
       path        => $::path,
       notify      => Service[$service_name]
     }
+  } elsif ($initstyle == 'upstart') {
```
Comment (deric) [inline manifests/service.pp]: It's called `upstart` but the script is in fact `sysvinit` or `init.d` script. That's quite confusing.

**Comment by deric** [comment]
What's the motivation for managing custom `init.d` script for ZooKeeper? There are service packages `zookeeperd` for Debian and `zookeeper-server` that provide similar functionality.

**Comment by Vlaszaty** [comment]
Motivation for managing your init.d in puppet instead of using the one that comes with the package are:

1. Puppet is about being able to control your application and enforce configuration. When you deploy a service with Puppet you want to be able to control all configuration in case you need certain configuration settings changed, also if that means that the default init script need to be adapted
2. The ability to install from different packages without having doubts about what exactly is happening in your init script, or the ability to install from packages that do not supply an init script

In my opinion, the init script is an integral part of the application configuration, so it should definitely be configurable through Puppet

**Comment by deric** [comment]
Sure, generally I try to avoid unnecessary complexity in module. Handling all possible combinations of versions, OS etc. is rather hard to test and maintain. Less code, means less problems.

However if certain OS does not support some options, contains bugs that are not fixed yet or just does not source some variable; it would be a good reason to override the defaults.

**Comment by deric** [comment]
I've changed names of parameters, in order to use this feature you should define:
```puppet
class { 'zookeeper':
  manage_service_file => true,
  service_provider    => 'init',
}
```

**Pull request merged**